src/test/java/com/commercetools/sync/sdk2/producttypes/utils/producttypeactionutils/BuildAttributeDefinitionUpdateActionsTest.java

```diff
@@ -0,0 +1,1516 @@
+  private static final String PRODUCT_TYPE_WITH_ATTRIBUTES_ADBC =
+      RES_ROOT + "product-type-with-attribute-definitions-adbc.json";
+  private static final String PRODUCT_TYPE_WITH_ATTRIBUTES_CBD =
+      RES_ROOT + "product-type-with-attribute-definitions-cbd.json";
+
+  private static final ProductTypeSyncOptions SYNC_OPTIONS =
+      ProductTypeSyncOptionsBuilder.of(mock(ProjectApiRoot.class)).build();
+
+  private static final AttributeDefinition ATTRIBUTE_DEFINITION_A =
+      createMockAttributeDefinitionBuilder()
+          .name("a")
+          .label(ofEnglish("label_en"))
+          .type(AttributeTypeBuilder::textBuilder)
+          .build();
+
+  private static final AttributeDefinition ATTRIBUTE_DEFINITION_B =
+      createMockAttributeDefinitionBuilder()
+          .name("b")
+          .label(ofEnglish("label_en"))
+          .type(AttributeTypeBuilder::textBuilder)
+          .build();
+
+  private static final AttributeDefinition ATTRIBUTE_DEFINITION_C =
+      createMockAttributeDefinitionBuilder()
+          .name("c")
+          .label(ofEnglish("label_en"))
+          .type(AttributeTypeBuilder::textBuilder)
+          .build();
+
+  private static final AttributeDefinition ATTRIBUTE_DEFINITION_D =
+      createMockAttributeDefinitionBuilder()
+          .name("d")
+          .label(ofEnglish("label_en"))
+          .type(AttributeTypeBuilder::textBuilder)
+          .isRequired(false)
+          .build();
+
+  @Test
+  void
+      buildAttributesUpdateActions_WithNullNewAttributesAndExistingAttributes_ShouldBuild3RemoveActions() {
+    final ProductType oldProductType =
+        readObjectFromResource(PRODUCT_TYPE_WITH_ATTRIBUTES_ABC, ProductType.class);
+
+    final List<ProductTypeUpdateAction> updateActions =
+        buildAttributesUpdateActions(
+            oldProductType,
+            ProductTypeDraftBuilder.of()
+                .key("key")
+                .name("name")
+                .description("description")
+                .attributes((List<AttributeDefinitionDraft>) null)
+                .build(),
+            SYNC_OPTIONS);
+
+    assertThat(updateActions)
+        .containsExactly(
+            ProductTypeRemoveAttributeDefinitionActionBuilder.of().name("a").build(),
+            ProductTypeRemoveAttributeDefinitionActionBuilder.of().name("b").build(),
+            ProductTypeRemoveAttributeDefinitionActionBuilder.of().name("c").build());
+  }
+
+  @Test
+  void
+      buildAttributesUpdateActions_WithNullNewAttributesAndNoOldAttributes_ShouldNotBuildActions() {
+    final ProductType oldProductType = mock(ProductType.class);
+    when(oldProductType.getAttributes()).thenReturn(emptyList());
+    when(oldProductType.getKey()).thenReturn("product_type_key_1");
+
+    final List<ProductTypeUpdateAction> updateActions =
+        buildAttributesUpdateActions(
+            oldProductType,
+            ProductTypeDraftBuilder.of()
+                .key("key")
```

Comment:
Shouldn't this be "product_type_key_1"?

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/989/commits/6b97efefa1d103aca2f9bbd274ccc99e0d84c77a